src/zps.c

```diff
@@ -15,439 +15,797 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include "zps.h"
-#include <stdio.h>
-#include <stdlib.h>
-#include <unistd.h>
-#include <getopt.h>
+#define _GNU_SOURCE
+
+#include <assert.h>
+#include <ctype.h>
+#include <dirent.h>
 #include <fcntl.h>
-#include <sys/stat.h>
-#include <string.h>
+#include <getopt.h>
+#include <limits.h>
```

Comment:
My LSP says `limits.h` is not used anywhere, can you check?

Reply:
Well, this include had to be used because we're using `PATH_MAX` for this, as long as it's defined in `<limits.h>`. If we were to use different filesystems from the `procfs`, we should rather not be doing it like this, as, actually, longer paths could still be used in theory. A probably more portable version for non-`procfs` purposes would be using `pathconf(3)` with `_PC_PATH_MAX`, but then we would need dynamic allocation with `malloc` or use funky stuff like variable-length arrays (not recommended).

Reply:
And also, the build with `musl` on the Alpine container failed because of the previously missing include.

Reply:
I see, then there is something funky going on with my LSP.